iceoryx_utils/include/iceoryx_utils/cxx/make_scoped_static.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
```

Comment:
This is wrong. It was created today.

I know the first implementation was in 2019 in the helplets.hpp file but thanks to git the whole history is preserved and correct. Here we have some implicit assumptions which just lead directly into a weird and complex world.

Reply:
Sorry but copyright does not rely on the Git history. This file was copied and pasted and not written by me. As a result all copyright information has to be copied to new file as well. It's done like this in every FOSS project I've worked on.

Reply:
@mossmaurice I strongly disagree. @elBoberido , @dkroenke , @budrus @FerdinandSpitzschnueffler @MatthiasKillat  what do you think?

The reason is the following. Let's assume you are right then everytime we do some refactoring and move a function into a new file we have to trace the whole function. This means when it was first written in file A, then moved to file B, then to file C and with the new refactoring to file D we have to trace the git history of the function through the files. In the new file we have then to add all the people who ever touched a piece of the function in the copyright header.

It is getting even worse when we would like to split up the function in the next refactoring in to different files. Then we have to trace every line!!! With every file we create and move code we have trace all the lines where they did come from! This is absolutely horrible.

Please do not close or resolve the conversation until we all agreed on how we proceed in such cases.

Reply:
@elfenpiff @mossmaurice 
Well, looking at the Eclipse Handbook could maybe clarify this: https://www.eclipse.org/projects/handbook/#legaldoc-faq

> How do file headers change when content is refactored?

> When you split a file during a refactoring exercise, any new files that you create must automatically inherit the header of the original. If even fragment of text from a single line of the original content survives the refactoring (whether generated, rote, or otherwise), then the original copyright statement and author information should be retained.

> Even a refactoring exercise that does not change behaviour, may add intellectual property. When refactoring adds content, copyright statements should be updated to reflect the addition. In concrete terms, you can either add the new copyright holders to the existing copyright statement or append "and others" (especially when the list of copyright holders gets long).

Reply:
Ok, thanks @dkroenke for clarifying this. If we should get into a situation with dozens of copyright holder we might switch to "and others". We can raise awareness about this topic in the next dev meetup.

Reply:
@mossmaurice @dkroenke How do we ensure the correctness of the copyright headers as a reviewer? Do we have to retrace the code lines and functions over the total git history? Or anyone else?

At the moment we can handle this somehow but when there will be more this is the ultimate time killer.